### PR TITLE
Trigger azure pipeline for multi gpu tests 

### PR DESCRIPTION
**Pull request by skyline75489**
The run on CircleCI:  https://app.circleci.com/pipelines/github/pytorch/pytorch/283891/workflows/e049872e-5327-4f8c-abc3-a72ca6a1a548/jobs/11462671

### CODE REVIEW

File: .circleci/regenerate.ps1 [hunk 1]
```diff
@@ -0,0 +1,5 @@
+cd $PSScriptRoot;
+$NewFile = New-TemporaryFile;
+python generate_config_yml.py > $NewFile.name
+(Get-Content $NewFile.name -Raw).TrimEnd().Replace("`r`n","`n") | Set-Content config.yml -Force
+Remove-Item $NewFile.name
```
Comment (malfet) [inline .circleci/regenerate.ps1:5]: Not sure  what are the PowerShell coding standarts, but please add newline here
```suggestion
Remove-Item $NewFile.name
```

File: .circleci/scripts/trigger_azure_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+s = requests.Session()
+s.headers.update({"Authorization": "Basic " + AZURE_DEVOPS_PAT_BASE64})
+
+build_base_url = AZURE_PIPELINE_BASE_URL + "_apis/build/builds?api-version=6.0"
+
+TARGET_BRANCH = os.environ.get("CIRCLE_BRANCH", "master")
+
+# Convert the branch name for Azure DevOps
+if TARGET_BRANCH.startswith('pull/'):
+    match = re.search(r'pull/(\d+)', TARGET_BRANCH)
```
Comment (malfet) [inline .circleci/scripts/trigger_azure_pipeline.py:21]: Let's make code a bit more error resilient
```suggestion
match = re.search(r'pull/(\d+)', TARGET_BRANCH)
if match is not None:
```

File: .circleci/scripts/trigger_azure_pipeline.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    pr_num = match.group(1)
+    SOURCE_BRANCH = 'refs/pull/' + str(pr_num) + '/head'
```
Comment (malfet) [inline .circleci/scripts/trigger_azure_pipeline.py:23]: It's ok to use Python3 features in codebase
```suggestion
    SOURCE_BRANCH = f'refs/pull/{pr_num}/head'
```

File: .circleci/scripts/trigger_azure_pipeline.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+else:
+    SOURCE_BRANCH = "refs/heads/" + TARGET_BRANCH
```
Comment (malfet) [inline .circleci/scripts/trigger_azure_pipeline.py:25]:
```suggestion
    SOURCE_BRANCH = f"refs/heads/{TARGET_BRANCH}"
```

**Review by malfet — APPROVED** [review]

**Comment by skyline75489** [comment]
@malfet Anything else I need to do to ship this?

**Comment by agolynski** [comment]
Seems like the workflow is failing, it doesn't provide much details and i cannot access build logs, e.g.
https://aiinfra.visualstudio.com/0628bce4-2d33-499e-bac5-530e12db160f/_apis/build/builds/148427/logs/3

**Pull request closed without merging**